### ISSUE TITLE
dynamic_rnn_decoder returns shape [?, batch_size, cell.output_size]

### ISSUE THREAD

**Issue by jbingel** [tensorflow/tensorflow#7702]
According to [the docs](https://www.tensorflow.org/api_docs/python/tf/contrib/seq2seq/dynamic_rnn_decoder),  the dynamic_rnn_decoder returns a tuple contaning `outputs`, which is a Tensor of shape `[max_time, batch_size, cell.output_size]`(provided `time_major==True`).

In my case, however, the first dimension of that Tensor is returned as underspecified (Dimension `?`), and in fact depending on the provided batch when running the RNN. 

If this is the intended behaviour, it should probably be highlighted in the documentation that `max_time` is variable.

Reproduce with:

```import tensorflow as tf
import numpy as np

# toy data, timesteps between 1 and 10
timesteps = np.random.randint(1, 11, [10])
X=np.random.randint(0, 20, [10,10,1])

batch_size = 2
max_ts = 10
inputs = tf.placeholder(tf.float32, 
                        (max_ts, batch_size, 1), name="X_in")

cell_fw = tf.contrib.rnn.LSTMCell(50)
cell_bw = tf.contrib.rnn.LSTMCell(50)
cell_dec = tf.contrib.rnn.LSTMCell(50)

seq_lens = tf.placeholder(tf.int32, batch_size, name="seq_lens")

enc_outputs, states = tf.nn.bidirectional_dynamic_rnn(
    cell_fw, cell_bw, inputs, time_major=True, sequence_length=seq_lens, dtype=tf.float32)

decoder_inp = tf.concat(enc_outputs, axis=2) 

attention_states = tf.zeros([batch_size, 1, cell_dec.output_size],
                                    name="attention_states")

att_keys, att_vals, att_score_fn, att_construct_fn = \
            tf.contrib.seq2seq.prepare_attention(attention_states,
                                                 attention_option="luong",
                                                 num_units=50)

dynamic_fn_train = tf.contrib.seq2seq.attention_decoder_fn_train(
            states[0], att_keys, att_vals, att_score_fn, att_construct_fn)

outputs, _, _ = tf.contrib.seq2seq.dynamic_rnn_decoder(
            cell_dec, dynamic_fn_train, decoder_inp, time_major=True,
            sequence_length=seq_lens)

with tf.Session() as sess:
    feed_dict = {inputs: X[:,:2,:], seq_lens: ts[:2]}
    sess.run(tf.global_variables_initializer())
    out = sess.run(outputs, feed_dict=feed_dict)
    print(out.shape[0])
```

The very last print statement will show that the first output dimension is not max_ts, but the max timestep of the batch (<= 10)

**Comment by ebrevdo** [comment]
Indeed, we should make it clear that dynamic_rnn will not know - at graph build time - the max_time value; that value is a Tensor whose value is tf.shape(inputs)[0] (0 when time_major=True, 1 when time_major=False).  PRs to improve the documentation are welcome!

**Comment by mckinziebrandon** [comment]
Anyone working on this? I've been using dynamic_rnn a lot lately and would be happy to write the requested documentation. I've been looking to start contributing here and this seems like a good issue to start with.

**Comment by ebrevdo** [comment]
Keep in mind that a lot of the functions in tf.contrib.seq2seq have been
rewritten in object-oriented style (on github master).  So for example:

tf.contrib.seq2seq.dynamic_rnn_decoder

and

tf.contrib.seq2seq.attention_decoder_fn_train

no longer exist.  a new attention mechanism is being pushed by next week.
 a new decoder already exists but is object oriented.

On Fri, Mar 10, 2017 at 10:33 AM, Brandon McKinzie <notifications@github.com

**Comment by mckinziebrandon** [comment]
Yes, I noticed that, but figured it may be useful to update the current docs for users not on the master branch. In any case, let me know if any help is needed for seq2seq documentation. I've been following the object-oriented code in master closely and I'd be happy to help with either.

Along the same lines, are there any other issues that would be good "starter" contributions? Not sure where the best place is for suggesting/asking such things so any info would be useful. I read the contribution guidelines, and they suggested that the comments here are the place to discuss. Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and this issue has an assignee.Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by ebrevdo** [comment]
Contributions welcome to the existing OO-based API.   cc += @qlzh727

**Comment by kumariko** [comment]
@jbingel Could you please let us know, if you still need help on this issue. Since contrib has been depreciated in Tensorflow 2.x, Please do upgrade to a latest Tensorflow version 2.8 . Attaching [migration](https://www.tensorflow.org/guide/migrate) guide for reference. Thanks!
